src/app/Common.DB.Postgres/Services/UsersService.cs

```diff
@@ -0,0 +1,146 @@
+        }
+
+        public Task MarkEmailAsVerifiedAsync(string id)
+        {
+            return _userRepository.GetUpdateQuery(id).Set(x => x.IsEmailVerified, true).UpdateAsync();
+        }
+
+        public Task UpdateLastRequestAsync(string id)
+        {
+            return _userRepository.GetUpdateQuery(id).Set(u => u.LastRequest, DateTime.UtcNow).UpdateAsync();
+        }
+
+        public Task UpdateResetPasswordTokenAsync(string id, string token)
+        {
+            return _userRepository.GetUpdateQuery(id).Set(u => u.ResetPasswordToken, token).UpdateAsync();
+        }
+
+        public Task UpdatePasswordAsync(string id, string newPassword)
+        {
+            return _userRepository.GetUpdateQuery(id)
+                .Set(user => user.PasswordHash, newPassword.GetHash())
+                .Set(user => user.ResetPasswordToken, string.Empty)
+                .UpdateAsync();
+        }
+
+        public Task UpdateInfoAsync(string id, string email, string firstName, string lastName)
+        {
+            return _userRepository.GetUpdateQuery(id)
+               .Set(user => user.Email, email)
+               .Set(user => user.FirstName, firstName)
+               .Set(user => user.LastName, lastName)
+               .UpdateAsync();
+        }
+
+        public async Task<IUser> CreateUserAccountAsync(CreateUserModel model)
+        {
+            var signUpToken = SecurityUtils.GenerateSecureToken();
+
+            var newUser = new User
+            {
+                FirstName = model.FirstName,
+                LastName = model.LastName,
+                PasswordHash = model.Password.GetHash(),
+                Email = model.Email,
+                IsEmailVerified = false,
+                SignupToken = signUpToken
+            };
+
+            await _userRepository.InsertAsync(newUser);
+
+            _emailService.SendSignUpWelcome(new SignUpWelcomeModel
+            {
+                Email = model.Email,
+                SignUpToken = signUpToken
+            });
+
+            return newUser;
+        }
+
+        public async Task<IUser> CreateUserAccountAsync(CreateUserGoogleModel model)
+        {
+            var newUser = new User
+            {
+                FirstName = model.FirstName,
+                LastName = model.LastName,
+                Email = model.Email,
+                IsEmailVerified = true,
+                OAuthGoogle = true
+            };
+
+            await _userRepository.InsertAsync(newUser);
+
+            return newUser;
+        }
+
+        public Task EnableGoogleAuthAsync(string id)
+        {
+            return _userRepository.GetUpdateQuery(id)
+               .Set(user => user.OAuthGoogle, true)
+               .UpdateAsync();
+        }
+
+        public async Task<bool> IsEmailInUseAsync(string? userIdToExclude, string email)
+        {
+            var usersQuery = _userRepository.GetQuery().Where(x => x.Email.ToUpper() == email.ToUpper());
```

Comment:
Can we use string.Equals + StringComparison?

Reply:
I believe no. Linq2db can't translate it to sql. It didn't work with EF Core either for me. I think we stuck with ToLower(). It seems we can't even cover it with an extension method. I think I saw tests that SQL Server likes ToUpper more so I used it. But it seems if there are more lowercase letters in the source then ToLower is more efficient. But I am not sure. It seems Postgres preferes lower() or ilike. And there is the "citext" column type what will do lower() under the hood. I couldn't reproduce "ilike" with linq2db.